ios/swift/JitsiSDKTest/src/ViewController.swift

```diff
@@ -58,28 +67,37 @@ class ViewController: UIViewController {
             // builder.audioMuted = true;
             // builder.videoMuted = true;
         }
-                
-        // setup view controller
-        let vc = UIViewController()
-        vc.modalPresentationStyle = .fullScreen
-        vc.view = jitsiMeetView
         
         // join room and display jitsi-call
         jitsiMeetView.join(options)
-        present(vc, animated: true, completion: nil)
         
+        // Enable jitsimeet view to be a view that can be displayed
+        // on top of all the things, and let the coordinator to manage
+        // the view state and interactions
+        pipViewCoordinator = PiPViewCoordinator(withView: jitsiMeetView)
+        pipViewCoordinator?.configureAsStickyView(withParentView: view)
+
+        // animate in
+        jitsiMeetView.alpha = 0
+        pipViewCoordinator?.show()
     }
     
     fileprivate func cleanUp() {
-        if(jitsiMeetView != nil) {
-            dismiss(animated: true, completion: nil)
-            jitsiMeetView = nil
-        }
+        jitsiMeetView?.removeFromSuperview()
+        jitsiMeetView = nil
+        pipViewCoordinator = nil
     }
 }
 
 extension ViewController: JitsiMeetViewDelegate {
+    
     func ready(toClose data: [AnyHashable : Any]!) {
-        cleanUp()
+        self.pipViewCoordinator?.hide() { _ in
```

Comment:
Doesn't this need to be posted to the main thread?

Reply:
ready(toClose data:) is called on the main thread

Reply:
👍 Can you also let Kyle from brave know?